pkg/controller/vsphere/session/session.go

```diff
@@ -153,7 +153,7 @@ func isValidUUID(str string) bool {
 
 func (s *Session) GetTask(ctx context.Context, taskRef string) (*mo.Task, error) {
 	if taskRef == "" {
-		return nil, nil
+		return nil, errors.New("taskRef can't be empty")
```

Comment:
I'm confused about the motivation of this change. Woudn't this result in a hot loop for e.g create
https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/vsphere/reconciler.go#L62

Reply:
how is this not breaking the unit test for Create()?

Reply:
We had taskRef always set in unit tests